File: 02 Set and Map/Vote/d62_q1a_vote.py
name2vote = {}
namecount = 0
N,K = [int(e) for e in input().split()]
for i in range(N):
    name = input()
    if(name not in name2vote):
        name2vote[name] = 1
        namecount += 1
    else:
        name2vote[name] += 1

if(namecount <= K):
    print(min(name2vote.values()))
else:
    vote2num = {}
    for name, vote in name2vote.items():
        if(vote not in vote2num):
            vote2num[vote] = 1
        else:
            vote2num[vote] += 1
    vote2num = sorted(vote2num.items())[::-1]
    
    total = 0
    for vote,num in vote2num:
        total += num
        if(total >= K):
            print(vote)
            break

File: 02 Set and Map/Vote/d62_q1a_vote.cpp
#include<iostream>
#include<vector>
#include<map>
#include<algorithm>
using namespace std;
int N,K;
map<string,int> name2vote;
vector<int> scores;
int main()
{
    // Input
    cin >> N >> K;
    while(N--)
    {
        // Input name
        string name;
        cin >> name;
        // Increase vote by 1
        if(name2vote.find(name) == name2vote.end()) name2vote[name] = 1;
        else name2vote[name] += 1;
    }
    // Loop get score in each item in name2vote and push_back to scores 
    for(auto &x : name2vote) { scores.push_back(x.second); }
    // Sort the scores
    sort(scores.begin(), scores.end());

    // Case 1: Candidates is less than K
    if(scores.size() < K) cout << scores[0];
    // Case 2: Candidates is more than K
    else cout << scores[scores.size() - K];
}